resolver/resolver.cc

```diff
@@ -2510,127 +2586,60 @@ class ResolveSignaturesWalk {
                 }
 
                 if (!lastSigs.empty()) {
-                    prodCounterInc("types.sig.count");
-
-                    auto loc = lastSigs[0]->loc;
                     if (ctx.file.data(ctx).originalSigil == core::StrictLevel::None &&
                         !lastSigs.front()->flags.isRewriterSynthesized) {
+                        auto loc = lastSigs.front()->loc;
                         if (auto e = ctx.beginError(loc, core::errors::Resolver::SigInFileWithoutSigil)) {
                             e.setHeader("To use `{}`, this file must declare an explicit `{}` sigil (found: "
                                         "none). If you're not sure which one to use, start with `{}`",
                                         "sig", "# typed:", "# typed: false");
                         }
                     }
 
-                    bool isOverloaded = lastSigs.size() > 1 && ctx.permitOverloadDefinitions(ctx.file);
-                    auto originalName = mdef.symbol.data(ctx)->name;
-                    if (isOverloaded) {
-                        ctx.state.mangleRenameSymbol(mdef.symbol, originalName);
-                    }
-                    int i = 0;
-
                     // process signatures in the context of either the current
                     // class, or the current singleton class, depending on if
                     // the current method is a self method.
                     core::SymbolRef sigOwner;
                     if (mdef.flags.isSelfMethod) {
-                        sigOwner = ctx.owner.data(ctx)->singletonClass(ctx);
+                        sigOwner = ctx.owner.data(ctx)->lookupSingletonClass(ctx);
```

Comment:
When do we create the singletonClass? (e.g., why do we expect it to already exist?)

Reply:
Great question. Let me verify that we explicitly create the expected singletons.

Reply:
`namer` creates it. It creates a singleton class for every class def, and ENFORCEs that all defined methods have an owner (and it handles the `self` case).

Reply:
I added an ENFORCE and a comment just in case